File: checkout-ruby/lib/item_value_discount.rb
require_relative "./promotional_rule"

class ItemValueDiscount < PromotionalRule

  attr_accessor :item_code

  def initialize(type, item, count)
    @value_type = type
    @item_code = item.product_code
    @item_count = count
  end

  def get_discount_entity_type
    RuleDiscountEntityType::ITEM
  end

  def get_rule_discount_entity
    item_count
  end

  private
  attr_accessor :item_count
end

File: checkout-java/src/main/java/com/vinterior/checkout/models/ItemValueDiscount.java
package com.vinterior.checkout.models;

import com.vinterior.checkout.RuleDiscountValue;
import com.vinterior.checkout.enums.RuleDiscountEntityType;
import lombok.Getter;

public class ItemValueDiscount extends PromotionalRule {

    @Getter
    private Integer itemCode;

    private int itemCount;

    public ItemValueDiscount(RuleDiscountValue ruleDiscountValue, Item item, int count) {
        super(ruleDiscountValue);
        itemCode = item.getProductCode();
        itemCount = count;
    }
    @Override
    public RuleDiscountEntityType getDiscountEntityType() {
        return RuleDiscountEntityType.ITEM;
    }

    @Override
    public double getRuleDiscountEntity() {
        return itemCount;
    }

}
